mrmustard/training/parametrized.py

```diff
@@ -40,8 +42,8 @@ class Parametrized:
 
     def __init__(self, **kwargs):  # NOTE: only kwargs so that we can use the arg names
         owner = f"{self.__class__.__qualname__}"
-
-        for name, value in kwargs.items():
+        self.param_names = {}
```

Comment:
In view of the previous comments made about the use of the `Parametrized` class as a container of the parameters, this can be promoted to a property that returns the parameter names and values.

Reply:
But we need to know the parameter names in the right order from somewhere. This is what parameter_names does.